[Do not merge] Nitro Stylus Diff

rachel-bousfield opened the pull request:
This temporary PR provides a diff between `nitro-contracts` and `stylus-contracts` to aid in review.

gzeoneth commented on src/osp/OneStepProofEntry.sol:
```diff
@@ -122,7 +126,23 @@ contract OneStepProofEntry is IOneStepProofEntry {
         (mach, mod) = prover.executeOneStep(execCtx, mach, mod, inst, proof);
 
-        mach.modulesRoot = modProof.computeRootFromModule(oldModIdx, mod);
+        bool updateRoot = !(opcode == Instructions.LINK_MODULE ||
+            opcode == Instructions.UNLINK_MODULE);
```
can simplify into
```suggestion
        bool updateRoot = (opcode != Instructions.LINK_MODULE) &&
            (opcode != Instructions.UNLINK_MODULE);
```

gzeoneth commented on src/state/ModuleMemory.sol:
```diff
@@ -11,6 +11,8 @@ import "./ModuleMemoryCompact.sol";
 library ModuleMemoryLib {
     using MerkleProofLib for MerkleProof;
 
+    uint256 private constant LEAF_SIZE = 32;
```
This is defined separately in multiple contracts, we should consider to consolidate them

gzeoneth commented on src/osp/OneStepProverHostIo.sol:
```diff
@@ -287,6 +290,128 @@ contract OneStepProverHostIo is IOneStepProver {
+        Module memory,
+        Instruction calldata,
+        bytes calldata proof
+    ) internal view {
```
```suggestion
        bytes calldata
    ) internal pure {
```

gzeoneth commented on src/precompiles/ArbOwner.sol:
```diff
@@ -1,89 +1,113 @@
+ **/
 interface ArbOwner {
-    /// @notice Add account as a chain owner
+    // @notice Add account as a chain owner
```
I believes natspec [expect](https://docs.soliditylang.org/en/develop/layout-of-source-files.html#comments) triple slash

gzeoneth reviewed (changes requested):
some minor suggestions, yet to review the logic against nitro

PlasmaPower committed:
Add KZG preimage proof

rachel-bousfield committed:
new ArbWasm methods

rachel-bousfield committed:
keepalive method

rachel-bousfield committed:
ArbOwner methods

PlasmaPower committed:
Clarify that we're iterating over bits

PlasmaPower committed:
Clarify that we're iterating over bits

rachel-bousfield committed:
docstrings & errors

Commit:
account_code checks

rachel-bousfield committed:
initGas

PlasmaPower committed:
Pull out modExp256 into function

PlasmaPower committed:
Pull out modExp256 into function

rachel-bousfield committed:
Merge pull request #103 from OffchainLabs/change-date-2028

2028 change date

rachel-bousfield committed:
Merge pull request #26 from OffchainLabs/change-date-2028

2028 change date

PlasmaPower committed:
Fix formatting

PlasmaPower committed:
Fix formatting

rachel-bousfield committed:
Merge pull request #28 from OffchainLabs/activation-pricing

Activation Pricing

gvladika committed:
Add unit test for fee token based chain

gvladika committed:
Add test for arbitrum hosted ETH-based chain

gvladika committed:
Make it more readable

gvladika committed:
Add missing deployment block

PlasmaPower committed:
Address some PR comments

PlasmaPower committed:
Address some PR comments

PlasmaPower committed:
Add additional comment explaining proof data

PlasmaPower committed:
Add additional comment explaining proof data

PlasmaPower committed:
Require blsModulus matches expected value

PlasmaPower committed:
Require blsModulus matches expected value

PlasmaPower committed:
Fix comparison in comment

PlasmaPower committed:
Fix comparison in comment

gzeoneth committed:
Merge branch 'develop' into seq-inbox-tests

gzeoneth committed:
Merge branch 'develop' into kzg-proof

gzeoneth committed:
fix: test account init

gzeoneth committed:
Merge pull request #95 from OffchainLabs/seq-inbox-tests

Added some basic seq inbox foundry tests

yahgwai committed:
Updated gas refunder for blob costs

Formatting

Updates for code review

Remove whitespace

Formatting

yahgwai committed:
Moved gas refund enabled to separate file to for interface compatibility with block basefee

gzeoneth committed:
refactor: remove sequencer immutable optimation

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into 4844-only-2

gzeoneth committed:
chore: remove dot file

gzeoneth committed:
Merge branch 'develop' into kzg-proof

gzeoneth committed:
Merge pull request #104 from OffchainLabs/kzg-proof

Add WAVM KZG preimage proof

yahgwai committed:
Added both block basefee and blob basefee to batch spending report

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into 4844-only

gzeoneth committed:
fix: add DelegateCallAware

hkalodner committed:
Update readme and license for AEP

hkalodner committed:
Merge pull request #113 from OffchainLabs/aep-update

Update readme and license for AEP

hkalodner committed:
Fix typo in BSL

hkalodner committed:
Fix AEP link

PlasmaPower committed:
Include blob cost in extraGas field of batch posting reports

PlasmaPower committed:
Expect extraGas field in batch posting report

gzeoneth committed:
chore: revert SequencerInboxStub changes

PlasmaPower committed:
Merge branch '4844-only' into blob-extra-gas

PlasmaPower committed:
Fix expected log in SequencerInboxTest

yahgwai committed:
Moved 4844 readers into a single yul contract - reduces stack size in the sequencer inbox

yahgwai committed:
Merge from 4844-only

yahgwai committed:
Formatting

yahgwai committed:
Formatting

gzeoneth committed:
test: fix

gzeoneth committed:
test: fix 4844

gzeoneth committed:
Merge branch '4844-only' into batch-poster-manager

gzeoneth committed:
fix: various

gzeoneth committed:
fix: set batch poster manager in rollup creator

gzeoneth committed:
fix: solhint

gzeoneth committed:
Merge pull request #114 from OffchainLabs/blob-extra-gas

Include blob cost in extraGas field of batch posting reports

gzeoneth committed:
Merge remote-tracking branch 'origin/4844-only' into 4844-gas-refunder

gzeoneth committed:
Merge pull request #111 from OffchainLabs/4844-gas-refunder

Blob data gas refunds

PlasmaPower committed:
Add safety checks to Reader4844.yul

PlasmaPower committed:
Use solc's formatting for Yul

gzeoneth committed:
Merge pull request #115 from OffchainLabs/reader-4844-safety

Add safety checks to Reader4844.yul

gvladika committed:
Instead of reporting 0 basefee, don't send report at all in case fee token is used

yahgwai committed:
Update src/bridge/SequencerInbox.sol

Co-authored-by: gzeon <hng@offchainlabs.com>

yahgwai committed:
Added postupgradeinit for the challenge manafer

yahgwai committed:
Formatting

yahgwai committed:
Added postupgradeinit tests

gvladika committed:
Make 'isUsingFeeToken' immutable to avoid nativeToken() calls

gvladika committed:
Add constructor/initialization unit tests

gvladika committed:
Add import

yahgwai committed:
Merge pull request #116 from OffchainLabs/4844-chalman-update

Added postupgradeinit for the challenge manager

gvladika committed:
Merge branch 'develop' into fee-token-reports

rachel-bousfield committed:
add datafee

rachel-bousfield committed:
add comment

rachel-bousfield committed:
Merge pull request #29 from OffchainLabs/add-datafee

Add activation datafee

rachel-bousfield committed:
add keepalive event

rachel-bousfield committed:
rename programLifetimeExtended

rachel-bousfield committed:
Merge pull request #30 from OffchainLabs/keepalive-event

Add keepalive event

PlasmaPower committed:
Use 0x50 as the header byte for 4844 batches

gzeoneth committed:
test: update DATA_BLOB_HEADER_FLAG

gzeoneth committed:
Merge pull request #118 from OffchainLabs/4844-0x50

gvladika committed:
Merge branch '4844-only' into fee-token-reports

gvladika committed:
Format and remove unnecessary

gvladika committed:
Format

PlasmaPower committed:
Support zero basefee for gas estimation

PlasmaPower committed:
Fix SequencerInboxStubCreator

gzeoneth committed:
Merge pull request #119 from OffchainLabs/zero-basefee

Support zero basefee for gas estimation

gzeoneth committed:
Merge pull request #120 from OffchainLabs/fix-stub-creator

Fix SequencerInboxStubCreator

gzeoneth committed:
Merge branch '4844-only' into fee-token-reports

gzeoneth committed:
fix: stub creator

yahgwai committed:
Removed TODOs

yahgwai committed:
Increased data size

gzeoneth committed:
fix: txorigin as spender

gzeoneth committed:
chore: deprecate old method

gzeoneth committed:
fix: duplicated declaration

gzeoneth committed:
refactor: move blob gas logic

gzeoneth committed:
format: fix

gzeoneth committed:
docs: more comment

gzeoneth committed:
docs: comments

gzeoneth committed:
Merge pull request #121 from OffchainLabs/4844-blob-spend

misc: spending report logic and interface

gzeoneth committed:
Merge pull request #109 from OffchainLabs/4844-only

feat: 4844 implementation

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into batch-poster-manager

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into fee-token-reports

gzeoneth committed:
fix: disable fee token blob spending report

gzeoneth committed:
Merge pull request #108 from OffchainLabs/fee-token-reports

Do not send batch poster reports in fee token based chains

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into batch-poster-manager

gzeoneth committed:
Merge pull request #74 from OffchainLabs/batch-poster-manager

Add a batch poster manager for the sequencer inbox

yahgwai committed:
Updated 4844 test for custom fee token arg

yahgwai committed:
Merge pull request #124 from OffchainLabs/missing-4844-flag

Missing custom fee token arg in 4844 tests

PlasmaPower committed:
Add missing getters for L1 pricing parameters to ArbGasInfo

PlasmaPower committed:
Add comments noting precompile methods are new to ArbOS 20

PlasmaPower committed:
Add ArbOS precompile method to get scheduled upgrade

PlasmaPower committed:
Fix formatting

PlasmaPower committed:
Merge pull request #126 from OffchainLabs/scheduled-upgrade-getter

Add ArbOS precompile method to get scheduled upgrade

gzeoneth committed:
Merge branch 'develop' into missing-l1-pricing-getters

tsahee committed:
machine: support multistack in hashes and proofs

tsahee committed:
remove error guard's enable

replaced by machine.cothread

tsahee committed:
fix introducing cothreads

tsahee committed:
add cothread onestepproofs

gzeoneth committed:
Merge pull request #125 from OffchainLabs/missing-l1-pricing-getters

Add missing getters for L1 pricing parameters to ArbGasInfo

rachel-bousfield committed:
Merge remote-tracking branch 'stylus/stylus' into hostio-code

rachel-bousfield committed:
improve spacing

Commit:
Merge branch 'main' into develop; fix license conflict

Commit:
Merge pull request #127 from OffchainLabs/develop-main

merge main into develop; fix license.md conflict

rachel-bousfield committed:
Merge pull request #27 from OffchainLabs/hostio-code

account_code checks

tsahee committed:
removing error guard

ganeshvanahalli committed:
Add contract to test pending block's time and number advance

ganeshvanahalli committed:
fix name

ganeshvanahalli committed:
refactor

gzeoneth committed:
chore: remove sol2uml

hkalodner committed:
Merge pull request #134 from OffchainLabs/remove-sol2uml

chore: remove sol2uml

tsahee committed:
fix cothread support, use recoveryPc

gzeoneth committed:
fix: yarn format and build warnings

Co-authored-by: shotaronowhere <shawtarohgn@gmail.com>

gzeoneth committed:
fix: solhint warnings

gzeoneth committed:
fix: eslint warnings

gzeoneth committed:
fmt: spacing

gzeoneth committed:
test: fix

gzeoneth committed:
test: unused var

gzeoneth committed:
chore: swap ignore

gzeoneth committed:
release: 1.2.0

gzeoneth committed:
Merge pull request #137 from OffchainLabs/fix-lint-warnings

fix: format and lint warnings

gzeoneth committed:
Merge remote-tracking branch 'origin/main' into develop

gzeoneth committed:
Merge pull request #138 from OffchainLabs/merge-main

chore: merge main into develop

gzeoneth committed:
refactor: deployment utils

gzeoneth committed:
deploy: 4844 upgrade

gzeoneth committed:
chore: add yul build to release

gvladika committed:
Add reader4844 to deployment script

gzeoneth committed:
Merge pull request #141 from OffchainLabs/scripts-blob-reader

Add reader4844 to deployment script

gzeoneth committed:
fix: struct change

gzeoneth committed:
Merge branch 'deploy-tools' into fix-struct-change

gvladika committed:
Properly set seq inbox blob reader

gvladika committed:
Update script to work with latest design

gzeoneth committed:
Merge pull request #143 from OffchainLabs/fix-deploy-params

Set proper params for SequencerInbox

Commit:
merge audit-ci changes onto develop

Commit:
update audit-ci json

Commit:
update arbrollup.spec with waffle v4 syntax

gvladika committed:
Add check for 4bytes function signatures

gvladika committed:
Add to CI

gvladika committed:
Update gitignore

gvladika committed:
Merge branch 'develop' into sig-ci-check

gvladika committed:
Typo

gvladika committed:
Merge branch 'sig-ci-check' of github.com:OffchainLabs/nitro-contracts into sig-ci-check

tsahee committed:
multistack: 0 for empty remaining

aligns with our stack hashing

tsahee committed:
lint

gzeoneth committed:
Merge pull request #140 from OffchainLabs/deploy-tools

chore: deployment helper

gzeoneth committed:
Merge pull request #139 from OffchainLabs/develop

release: v1.2.0

gzeoneth committed:
Merge branch 'develop' into fix-struct-change

gzeoneth committed:
Merge branch 'develop' into sig-ci-check

gzeoneth committed:
Merge pull request #145 from OffchainLabs/sig-ci-check

Check function signatures in CI

gzeoneth committed:
Merge remote-tracking branch 'origin/develop' into fix-struct-change

gzeoneth committed:
chore: 4bytes update

gzeoneth committed:
chore: remove unused errors

gzeoneth committed:
feat: safe cast in setMaxTimeVariation

gzeoneth committed:
fmt: yarn format

gzeoneth committed:
test: setMaxTimeVariation

gzeoneth committed:
fix: eslint

gzeoneth committed:
v1.2.1

gzeoneth committed:
Merge pull request #142 from OffchainLabs/fix-struct-change

fix: struct change

Commit:
Merge branch 'develop' into audit-ci-dev, fix conflict

gzeoneth committed:
Merge pull request #146 from OffchainLabs/develop

release: v1.2.1

gzeoneth committed:
Merge pull request #144 from OffchainLabs/audit-ci-dev

add audit ci; update / allowlist dependencies; update waffle tests for v4 [v2]

ganeshvanahalli committed:
Merge branch 'develop' into add-pendingblk-blktimeadvance-contract

gvladika committed:
Basic Slither CI config

gvladika committed:
Use sarif

gvladika committed:
Check if job fails

gvladika committed:
Do not fail

gvladika committed:
Add slither config

gvladika committed:
Exclude optimizations

gvladika committed:
Exclude test/mock files

gvladika committed:
Upload triage DB

gvladika committed:
Add dangerous call to test slither reporter

gvladika committed:
Revert "Add dangerous call to test slither reporter"

This reverts commit ccc3ff2ac315881d922ad1c0a4a94b1944adb59a.

gvladika committed:
Update config to always upload results

gzeoneth committed:
chore: remove hardhat patch

gzeoneth committed:
fix: foundry optimizer_runs

ganeshvanahalli committed:
Merge pull request #133 from OffchainLabs/add-pendingblk-blktimeadvance-contract

gzeoneth committed:
Merge branch 'remove-hardhat-patch' into fix-audit

gzeoneth committed:
fix: audit:ci

gzeoneth committed:
ci: unignore advisories

gzeoneth committed:
Merge pull request #152 from OffchainLabs/fix-audit

fix: audit ci

Commit:
update yarn audit: OZ base 64

gzeoneth committed:
Merge pull request #157 from OffchainLabs/oz-audit

update yarn audit: OZ

gzeoneth committed:
Merge branch 'develop' into slither-integration

gvladika committed:
Update triage slither db

INFO:Detectors:
0: GasRefundEnabled.refundsGas(IGasRefunder,IReader4844) (src/libraries/GasRefundEnabled.sol#17-51) performs a multiplication on the result of a division:
	- calldataWords = (calldataSize + 31) / 32 (src/libraries/GasRefundEnabled.sol#22)
	- startGasLeft += calldataWords * 6 + (calldataWords ** 2) / 512 (src/libraries/GasRefundEnabled.sol#24)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#divide-before-multiply
Results to hide during next runs: "0,1,...,1" or "All" (enter to not hide results):
All
INFO:Detectors:
0: Multiple retryable tickets created in the same function:
	 -_fundAndDeploy(_inbox,NICK_CREATE2_VALUE,NICK_CREATE2_DEPLOYER,NICK_CREATE2_PAYLOAD,isUsingFeeToken,_maxFeePerGas) (src/rollup/DeployHelper.sol#96-103)
	 -_fundAndDeploy(_inbox,NICK_CREATE2_VALUE,NICK_CREATE2_DEPLOYER,NICK_CREATE2_PAYLOAD,isUsingFeeToken,_maxFeePerGas) (src/rollup/DeployHelper.sol#96-103)

gzeoneth committed:
Merge pull request #148 from OffchainLabs/slither-integration

Add Slither to CI

rachel-bousfield committed:
fix typo

rachel-bousfield committed:
Merge pull request #32 from OffchainLabs/go-wasi-edits

Go-wasi edits

rachel-bousfield committed:
Merge pull request #31 from OffchainLabs/cothread_support

Cothread support

tsahee committed:
Merge commit '9a6bfad2363322099d399698751551ff044c7a72' into nitro-2.2.5-merge

ganeshvanahalli committed:
Add function to test if arbBlockHash on pending block returns latest

ganeshvanahalli committed:
address PR comments

rachel-bousfield committed:
Merge pull request #34 from OffchainLabs/nitro-2.2.5-merge

Nitro 2.2.5 merge

rachel-bousfield committed:
cached init gas

tsahee committed:
Merge commit '1cab72ff3' into merge-nitro-v2.3.4

rachel-bousfield committed:
min cached init gas

rachel-bousfield committed:
ArbWasmCache

rachel-bousfield committed:
codehashIsCached precompile method

rachel-bousfield committed:
simple cache manager

rachel-bousfield committed:
precompile list method & events

rachel-bousfield committed:
remove trie table

rachel-bousfield committed:
add actual impl

rachel-bousfield committed:
format

rachel-bousfield committed:
32-bit asm size

rachel-bousfield committed:
better idioms & API

rachel-bousfield committed:
add events and simplify

rachel-bousfield committed:
Merge pull request #35 from OffchainLabs/merge-nitro-v2.3.4

Merge nitro v2.3.4 (rc3)

rachel-bousfield committed:
Merge remote-tracking branch 'stylus/stylus' into init-cache

rachel-bousfield committed:
fix condition

rachel-bousfield committed:
fix comparison

rachel-bousfield committed:
txCacheSize methods

rachel-bousfield committed:
add test case

rachel-bousfield committed:
add makeSpace

rachel-bousfield committed:
math hostio test

rachel-bousfield committed:
Merge pull request #38 from OffchainLabs/math-hostios

Math Host I/Os

rachel-bousfield committed:
Merge branch 'stylus' into init-cache

rachel-bousfield committed:
limit makeSpace

rachel-bousfield committed:
return amount of space left

rachel-bousfield committed:
Merge pull request #36 from OffchainLabs/init-cache

Stylus Cache Manager

rachel-bousfield committed:
optimize func merkle

rachel-bousfield committed:
fix root proof

rachel-bousfield committed:
O(n) instruction hashing

rachel-bousfield committed:
Merge pull request #39 from OffchainLabs/optimize-func-merkle

Optimize Function Merklization

rachel-bousfield committed:
address review comments

rachel-bousfield committed:
remove '=' in pin

rachel-bousfield committed:
Merge pull request #41 from OffchainLabs/cache-manager-feedback

Cache Manager Tweaks
